### PR TITLE
SpaceX Project

### PR DESCRIPTION
**Pull request by er-contreras**
- Config & basic setup
- Redux: Fetch data and update Redux store
- Render UI:lists
- Redux: Write actions and reducers for booking rockets and joining missions
- Render UI: My Profile section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,78 @@
...
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+## Authors
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+👤 **Christian Erick Contreras**
 
-### Code Splitting
+- GitHub: [@eri8-9](https://github.com/eri8-9)
+- Linkedin: [LinkedIn](https:linkedin.com/in/christian-erick-contreras-9945b820b)
```
Comment (MahmoudBakr23) [inline README.md]: - [x] Kindly make sure to provide a valid link to your LinkedIn account 😄 🙏 This is for a completely professional README file. 💯

File: src/components/rockets.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+              onClick={() => {
+                reserved = rocket.reserved;
+                if (!reserved) {
+                  reserveRocketHandler(rocket.id);
+                } else {
+                  cancelReservationHandler(rocket.id);
+                }
+              }}
+            >
+              Reserve Rocket
```
Comment (MahmoudBakr23) [inline src/components/rockets.js]: - [x] In this button, please change the text of it when it's on the cancelation process to follow the given design and also to make more sense 😄 🙏 You could've done it as the missions' buttons or you could write a very simple js function that changes the text of the button 💯 
_Please check the image in the project requirements right [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#:~:text=My%20Profile%20section.-,Rockets%20/Dragons,-The%20Rockets%20section)._

File: src/assets/styles/missions.css [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  color: red;
+}
+
+.reserve_btn:hover {
+  background-color: transparent;
+  font-size: 1rem;
+  padding: 10px;
+  color: black;
+  border: 1px solid #c0c0c0;
+}
```
Comment (MahmoudBakr23) [inline src/assets/styles/missions.css]: - [x] Kindly follow the CSS best practices and combine all the selectors that share the same rules. 😄 🙏 This is for cleaner and more optimized code. 💯 
_You can find about this tip and other useful ones right [here](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md)._
>Kindly apply this request to all similar situations through your files. 🙏

File: README.md [hunk 4]
```diff
@@ -1,70 +1,78 @@
...
+- Redux.
+- JavaScript.
+- SpaceX api
 
-Launches the test runner in the interactive watch mode.\
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+## Video
 
-### `npm run build`
+[Link](https://www.loom.com/share/a451a571209c414b8663b7d8a2a812ba) to video on Loom.
```
Comment (MahmoudBakr23) [inline README.md]: - [x] Good job presenting your capstone 👏 However, I believe that @eri8-9 voice doesn't seem to be obvious at all! 🙂 So, I'm afraid you'll have to record another presentation video 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello team 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct
- Workflow is correct
- Descriptive PR headline, summary, and commit messages
- Great use of React hooks along with Redux features
- Design followed

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [x] Please write unit tests using jest and react library as mentioned in the project requirements. Currently, there are no tests! 🙏

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/apiManager.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const spaceXAPI = {
+  getRocketsData() {
+    const rockets = fetch('https://api.spacexdata.com/v3/rockets')
+      .then((rawResponse) => rawResponse.json())
+      .then((response) => response);
+
+    return rockets;
+  },
+};
```
Comment (oxenprogrammer) [inline src/components/apiManager.js:9]: **Optional Change**
- [ ] Let us try to use the `async await` syntactic sugar for promises. It just makes the code more readable than chaining using the `.then` keyword

File: src/redux/missions/missions.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+});
+
+export const setMission = () => async (dispatch) => {
+  fetch(url)
+    .then((rawResponse) => rawResponse.json())
+    .then((response) => dispatch({
+      type: SET_MISSION,
+      payload: missionArray(response),
+    }));
+};
```
Comment (oxenprogrammer) [inline src/redux/missions/missions.js:35]: **Optional Change**
- [ ] Fair effort on using the `async await`  promise syntactic sugar. However, you're mixing two syntax here, and the `async` keyword in this context is redundant. May we please remove the `.then` and stick to the `async await` format.

File: src/redux/rockets/rocketsReducer.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  spaceXAPI.getRocketsData().then((res) => {
+    dispatch(getRocketsAction(
+      res.map((rocket) => ({
+        id: rocket.id,
+        name: rocket.rocket_name,
+        description: rocket.description,
+        flickr_images: rocket.flickr_images[0],
+      })),
+    ));
+  });
```
Comment (oxenprogrammer) [inline src/redux/rockets/rocketsReducer.js:47]: **Optional Change**
- [ ] Let us try to use the `async await` syntactic sugar for promises. It just makes the code more readable than chaining using the `.then` keyword

**Review by oxenprogrammer — APPROVED** [review]
Hi **@eri8-9** and **@GuilhermeGiachellin** , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**